crates/iceberg/src/expr/visitors/inclusive_projection.rs

```diff
@@ -0,0 +1,371 @@
+    }
+
+    fn not_null(
+        &mut self,
+        reference: &BoundReference,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn is_nan(
+        &mut self,
+        reference: &BoundReference,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn not_nan(
+        &mut self,
+        reference: &BoundReference,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn less_than(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn less_than_or_eq(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn greater_than(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn greater_than_or_eq(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn eq(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn not_eq(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn starts_with(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn not_starts_with(
+        &mut self,
+        reference: &BoundReference,
+        _literal: &Datum,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn r#in(
+        &mut self,
+        reference: &BoundReference,
+        _literals: &FnvHashSet<Datum>,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+
+    fn not_in(
+        &mut self,
+        reference: &BoundReference,
+        _literals: &FnvHashSet<Datum>,
+        predicate: &BoundPredicate,
+    ) -> crate::Result<Self::T> {
+        self.get_parts(reference, predicate)
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
It would be great to add some more tests here to at least cover all the transforms. In PyIceberg we have quite a nice suite: https://github.com/apache/iceberg-python/blob/62b527e74cff328fde5faa848519c56748a244c1/tests/expressions/test_projection.py

Reply:
I've added tests for the two remaining transforms that have been implemented in iceberg-rust that were untested (`Bucket` and `Truncate`).